packages/editor/src/browser/editor-quick-open.ts

```diff
@@ -0,0 +1,151 @@
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { inject, injectable } from 'inversify';
+import {
+    QuickOpenModel, QuickOpenItem, QuickOpenMode, PrefixQuickOpenService,
+    OpenerService, QuickOpenGroupItemOptions, QuickOpenItemOptions,
+    QuickOpenHandler, QuickOpenOptions
+} from '@theia/core/lib/browser';
+import { MessageService } from '@theia/core/lib/common/message-service';
+import URI from '@theia/core/lib/common/uri';
+import { LabelProvider } from '@theia/core/lib/browser/label-provider';
+import { Command } from '@theia/core/lib/common';
+import { CancellationTokenSource } from '@theia/core/lib/common';
+import { EditorManager } from './editor-manager';
+import { EditorWidget } from './editor-widget';
+
+export const SHOW_ALL_OPENED_EDITORS: Command = {
+    id: 'workbench.action.showAllEditors',
+    category: 'View',
+    label: 'Show All Opened Editors'
+};
+
+@injectable()
+export class EditorQuickOpenService implements QuickOpenModel, QuickOpenHandler {
+
+    @inject(OpenerService)
+    protected readonly openerService: OpenerService;
+
+    @inject(PrefixQuickOpenService)
+    protected readonly prefixQuickOpenService: PrefixQuickOpenService;
+
+    @inject(MessageService)
+    protected readonly messageService: MessageService;
+
+    @inject(LabelProvider)
+    protected readonly labelProvider: LabelProvider;
+
+    @inject(EditorManager)
+    protected readonly editorManager: EditorManager;
+
+    readonly prefix: string = 'edt ';
+
+    get description(): string {
+        return 'Show All Opened Editors';
+    }
+
+    getModel(): QuickOpenModel {
+        return this;
+    }
+
+    getOptions(): QuickOpenOptions {
+        return {
+            placeholder: '',
+            fuzzyMatchLabel: {
+                enableSeparateSubstringMatching: true
+            },
+            fuzzyMatchDescription: {
+                enableSeparateSubstringMatching: true
+            },
+            onClose: () => {
+                this.cancelIndicator.cancel();
+            }
+        };
+    }
+
+    open(): void {
+        this.prefixQuickOpenService.open(this.prefix);
+    }
+
+    private cancelIndicator = new CancellationTokenSource();
+
+    public async onType(lookFor: string, acceptor: (items: QuickOpenItem[]) => void): Promise<void> {
+        this.cancelIndicator.cancel();
+        this.cancelIndicator = new CancellationTokenSource();
+        const token = this.cancelIndicator.token;
+
+        const editorItems: QuickOpenItem[] = [];
+
+        // Get URIs of all EditorWidgets that are currently opened, sorted by URI
+        const widgets: URI[] = this.editorManager.getAllTrackedEditors()
```

Comment:
What is the expectations? Should we show all editors in history i.e. diff, git revision editors? For now the logic looks quirky.

- `getResourceUri` returns an underlying file URI for any editor URI (so for all editors)
- but below you filter by `file` scheme nevertheless

I'm not sure what you are trying to achieve.

Reply:
The idea was to get all editors that have URI (so they can be run) and are of `file` type, which is what `file-search` does (to exclude git-history files). I was trying to be consistent with the logic. But probably git-history files should not be filtered out when we show all opened editors.

Reply:
I will remove the "filtering by file scheme" logic.